### ISSUE TITLE
Segmentedtree UpdateTrees() vs UpdateTreesInBackground()

### ISSUE THREAD

**Issue by swharden** [ScottPlot/ScottPlot#705]
Hi @StendProg, in #692 @jl0pd commented about this method:

https://github.com/ScottPlot/ScottPlot/blob/2b1573713d1156bc155e4fdf410092d1be2cbc1b/src/ScottPlot/Plottable/DataStructures/Segmentedtree.cs#L54-L60

@jl0pd said we are not awaiting real operation, but only the creation of a task. By doing this users may begin to work data which is not yet updated properly. This solution was offered: 

```cs
public async Task SetSourceAsync(T[] data)
{
    sourceArray = data ?? throw new ArgumentNullException("Data cannot be null");
    await Task.Run(() => UpdateTrees());
}
```

This makes sense to me, as `UpdateTreesInBackground()` is just `Task.Run(() => { UpdateTrees(); })`, but before making this change I wanted to double check and make sure it looks good to you too. @StendProg, what do you think?

**Comment by StendProg** [comment]
Hi @swharden,
The funny thing is that this method has 0 references, since the user interacts with `SegmentedTree` through the use of `SignalPlot Api`, this method can be changed as desired, it will never be called.

The proposed change is more correct. After await for this method (with fixes), `SegmentedTree` is guaranteed to be fully updated.

As for the incorrect display of data, this is extremely unlikely, when source update run, the `TreesReady flag` is cleared first, and at the same time the render switches to a slow linear algorithm that does not use the `SegmentedTree` in any way (already updated array only used for Render).
Simply put, for the duration of the update, the` SegmentedTree` uses a slower algorithm that does not use invalid data.

**Comment by swharden** [comment]
It seems this method is unused by the library (so this change will have no practical effect), and the old code was very unlikely to produce a bug anyway. However, since it's a public method I went ahead and fixed it as suggested.

Thank you for your input @StendProg!
